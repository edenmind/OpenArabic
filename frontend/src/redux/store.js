import {
  arabicSentenceReducer,
  arabicTextReducer,
  arabicWordsReducer,
  authorReducer,
  categoryReducer,
  englishSentenceReducer,
  englishTextReducer,
  englishWordsReducer,
  sourceReducer,
  textReducer,
  wordByWordReducer,
} from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    title: textReducer,
    category: categoryReducer,
    author: authorReducer,
    source: sourceReducer,
    englishText: englishTextReducer,
    arabicText: arabicTextReducer,
    arabicWords: arabicWordsReducer,
    englishWords: englishWordsReducer,
    arabicSentence: arabicSentenceReducer,
    englishSentence: englishSentenceReducer,
    wordByWord: wordByWordReducer,
  },
})
