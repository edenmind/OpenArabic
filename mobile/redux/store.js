import { configureStore, combineReducers } from '@reduxjs/toolkit'

import { categoriesReducer, textReducer, textsReducer, UIStateReducer, wordsReducer } from './reducers.js'

const rootReducer = combineReducers({
  categories: categoriesReducer,
  text: textReducer,
  texts: textsReducer,
  textLoading: UIStateReducer,
  textsLoading: UIStateReducer,
  arabicFontSize: UIStateReducer,
  englishFontSize: UIStateReducer,
  isTransliterationOn: UIStateReducer,
  arabicFontName: UIStateReducer,
  practicingWords: UIStateReducer,
  words: wordsReducer,
  isDarkMode: UIStateReducer
})

const store = configureStore({
  reducer: rootReducer
})

export { store }
