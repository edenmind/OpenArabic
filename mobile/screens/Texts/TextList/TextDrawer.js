import React, { useState, useEffect } from "react";
import { StyleSheet } from "react-native";
import { TextListScreen } from "./TextListScreen";
import { createDrawerNavigator } from "@react-navigation/drawer";
import axios from "axios";

const Drawer = createDrawerNavigator();

export function TextDrawer() {
  const [categories, setCategories] = useState([{ categoryId: 0, name: "Adab" }]);
  const categoryItems = categories.map((category) => (
    <Drawer.Screen name={category.name} component={TextListScreen} initialParams={{ category: category.name }} options={{ title: category.name }} key={category.name} />
  ));

  useEffect(() => {
    axios.get("https://api.openarabic.io/api/categories").then(function (response) {
      setCategories(response.data);
    });
  }, [categoryItems]);

  return <Drawer.Navigator initialRouteName="Texts">{categoryItems}</Drawer.Navigator>;
}

export const style = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
});
