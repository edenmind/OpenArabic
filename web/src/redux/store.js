import * as reducers from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    text: reducers.textReducer,
  },
})
