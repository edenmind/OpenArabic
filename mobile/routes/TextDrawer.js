import { Caption, Title } from 'react-native-paper'
/* eslint-disable import/namespace */
import {
  DrawerContentScrollView,
  DrawerItemList,
  createDrawerNavigator
} from '@react-navigation/drawer'
import React, { Fragment } from 'react'
import { useDispatch, useSelector } from 'react-redux'

import COLORS from '../constants/colors'
import SCREENS from '../constants/screens'
import { StyleSheet } from 'react-native'
import TextList from '../screens/Category'
import { getCategories } from '../services/ApiService'
import { useFocusEffect } from '@react-navigation/core'

export default function TextDrawer() {
  const Drawer = createDrawerNavigator()
  const selector = (state) => state.categories
  const { categories } = useSelector(selector)
  const dispatch = useDispatch()

  const style = StyleSheet.create({
    info: {
      bottom: 10,
      left: 0,
      margin: 16,
      position: 'absolute'
    },
    title: {
      fontWeight: 'bold',
      marginBottom: 10,
      marginLeft: 17
    }
  })

  const categoryItems = categories.map((category, index) => (
    <Drawer.Screen
      name={category.name}
      key={index}
      component={TextList}
      initialParams={{ category: category.id }}
      options={{
        title: category.name,
        headerShown: true,
        drawerLabel: category.name
      }}
    />
  ))

  useFocusEffect(
    React.useCallback(() => {
      dispatch(getCategories())
    }, [dispatch])
  )

  const version = 'Version 1.2.0'
  const title = 'OpenArabic'

  function CustomDrawerContent(props) {
    return (
      <Fragment>
        <DrawerContentScrollView {...props}>
          <Title style={style.title}>{title}</Title>

          <DrawerItemList {...props} />
        </DrawerContentScrollView>

        <Caption style={style.info}>{version}</Caption>
      </Fragment>
    )
  }

  const homeScreen = (
    <Drawer.Screen
      name="All"
      component={TextList}
      initialParams={{ category: 'All' }}
      options={{
        title: SCREENS.home,
        headerShown: true,
        drawerLabel: 'Home'
      }}
      key={'999'}
    />
  )

  return (
    <Drawer.Navigator
      drawerContent={(props) => <CustomDrawerContent {...props} />}
      screenOptions={{
        unmountOnBlur: true,
        drawerStyle: {
          width: 150
        },
        headerStyle: {
          backgroundColor: COLORS.shinyOlive
        },
        headerTintColor: COLORS.darkOlive,
        drawerActiveBackgroundColor: COLORS.shinyOlive
      }}>
      {homeScreen}
      {categoryItems}
    </Drawer.Navigator>
  )
}
