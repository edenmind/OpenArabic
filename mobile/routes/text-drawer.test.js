import { describe, it, jest } from '@jest/globals'
import ShallowRenderer from 'react-test-renderer/shallow' // ES6
import React from 'react'
import TextDrawer from './text-drawer.js'
import { store } from '../redux/store.js'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'

jest.useFakeTimers()

describe('<TextDrawer />', () => {
  it('renders without crashing', () => {
    // @ts-ignore
    const renderer = new ShallowRenderer()
    renderer.render(
      <Provider store={store}>
        <NavigationContainer>
          <TextDrawer />
        </NavigationContainer>
      </Provider>
    )
  })
})
