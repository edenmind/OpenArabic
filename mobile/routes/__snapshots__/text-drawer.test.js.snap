// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextDrawer /> matches snapshot 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ForwardRef(NavigationContainerInner)>
    <TextDrawer />
  </ForwardRef(NavigationContainerInner)>
</Context.Provider>
`;
